irobotclient/test/test_main.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+from unittest.mock import MagicMock
+
+import requests
+import os
+
+import irobotclient.main
+
+
+class TestMain(unittest.TestCase):
+
+    test_data_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdata/')
+    output_dir = os.path.join(os.path.dirname(__file__), '..', '..', 'testdir/')
+
+    def setUp(self):
+        self._old_request_head = requests.head
+        self._response = requests.Response()
+        requests.head = MagicMock(spec=requests.head)
```

Comment:
For unit testing you don't need (or want) a working web server; that's reserved for integration testing.

What I believe @colin-nolan means here is that, if you had created an abstraction for `requests` (see previous comments), you could have just subclassed a dummy version of that and passed in an instantiation of it, rather than having to mock the raw `requests` library. That dummy version would essentially be a mock, in a "moral" sense, but because you'd control the interface -- you built it, after all -- it would be easier to manage without having to go into the details of how `requests` works. "Easier" implies "less potential for error".